src/libs/ui/settingsdialog.cpp

```diff
@@ -70,6 +72,13 @@ SettingsDialog::~SettingsDialog()
     delete ui;
 }
 
+void SettingsDialog::on_showToolsMenu_clicked()
+{
+    this->saveSettings();
+    this->reject();
+    m_application->showToolsMenu();
```

Comment:
This breaks encapsulation, you need to fire a signal instead of calling a method directly. And then simply connect to that signal in `MainWindow`.